app/views/helpers/NotificationErrorHelper.scala

```diff
@@ -0,0 +1,112 @@
+/*
+ * Copyright 2021 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package views.helpers
+
+import config.featureFlags.ChangeErrorLinkConfig
+import connectors.CodeListConnector
+import controllers.routes
+import models.declaration.notifications.NotificationError
+import models.ExportsDeclaration
+import play.api.i18n.Messages
+import play.twirl.api.{Html, HtmlFormat}
+import views.components.gds.ActionItemBuilder._
+import views.html.components.gds.{heading, _}
+import uk.gov.hmrc.govukfrontend.views.viewmodels.summarylist._
+import uk.gov.hmrc.govukfrontend.views.Aliases.Text
+import uk.gov.hmrc.govukfrontend.views.viewmodels.content.HtmlContent
+
+import javax.inject.{Inject, Singleton}
+
+@Singleton
+class NotificationErrorHelper @Inject()(
+  codeListConnector: CodeListConnector,
+  changeErrorLinkConfig: ChangeErrorLinkConfig,
+  heading: heading,
+  paragraphBody: paragraphBody
+) {
+  def formattedErrorDescription(notificationError: NotificationError)(implicit messages: Messages): List[Html] = {
+    val description = codeListConnector
+      .getDmsErrorCodesMap(messages.lang.toLocale)
+      .get(notificationError.validationCode)
+      .map(dmsErrorCode => dmsErrorCode.description)
+      .getOrElse(messages("error.unknown"))
+
+    if (description.trim.isEmpty)
+      List(Text("").asHtml)
+    else {
+      val desc = description.last match {
+        case '.' | '?' | '!'           => description
+        case _ if description.nonEmpty => s"$description."
+        case _                         => ""
+      }
+
+      desc
+        .split("\n")
+        .toList
+        .map(text => List(Text(text).asHtml, Html("<br>")))
+        .flatten
+        .filter(_.body.nonEmpty)
+        .dropRight(1)
+    }
+  }
+
+  def errorChangeAction(notificationError: NotificationError, declaration: ExportsDeclaration)(implicit messages: Messages) =
+    if (changeErrorLinkConfig.isEnabled) {
+      PointerHelper
+        .getChangeLinkCall(notificationError.pointer, declaration)
+        .fold(Some(Actions(items = Seq.empty))) { call =>
+          val errorPattern = notificationError.pointer.map(_.pattern).getOrElse("")
```

Comment:
Even if you need to pass a few params, I would still move this block (lines 72-81) to a `def`, in order to help the reader.